src/schemas.js

```diff
@@ -171,3 +170,247 @@ export const atomicSwapCreateSchema = {
     optional: true,
   },
 };
+
+export const txBuildSchema = {
+  outputs: {
+    in: ['body'],
+    errorMessage: 'Invalid outputs array',
+    isArray: true,
+    notEmpty: true,
+    optional: false,
+  },
+  'outputs.*.address': {
+    in: ['body'],
+    errorMessage: 'Invalid address',
+    isString: true,
+    optional: true,
+  },
+  'outputs.*.value': {
+    in: ['body'],
+    errorMessage: 'Invalid value',
+    isInt: {
+      options: {
+        min: 1,
+      },
+    },
+    toInt: true,
+    optional: true,
+  },
+  'outputs.*.token': {
+    in: ['body'],
+    errorMessage: 'Invalid token uid',
+    isString: true,
+    optional: true,
+    custom: {
+      // Test if token uid is a 64 character hex string or 00
+      options: value => /^(00|[0-9a-fA-F]{64})$/.test(value),
+    },
+  },
+  'outputs.*': {
+    in: ['body'],
+    isObject: true,
+    custom: {
+      options: value => {
+        if ('type' in value && value.type === 'data') {
+          if ('data' in value && !('token' in value)) {
+            return (/^[0-9a-fA-F]+$/.test(value.data));
+          }
+          return false;
+        }
+        // This is an address output
+        // Check that we have all params
+        return ('address' in value && 'value' in value);
+      },
+    },
+  },
+  inputs: {
+    in: ['body'],
+    errorMessage: 'Invalid inputs array',
+    isArray: true,
+    notEmpty: true,
+    optional: true,
+  },
+  change_address: {
+    in: ['body'],
+    errorMessage: 'Invalid change address',
+    isString: true,
+    optional: true,
+  },
+};
+
+export const txInputSchema = {
```

Comment:
Shouldn't this have the `inputs` array definition in the beginning? Or you don't need that?

```
inputs: {
    in: ['body'],
    errorMessage: 'Invalid inputs array',
    isArray: true,
    notEmpty: true,
  },
```

Reply:
We have 2 chains of validation, the `txBuildSchema` and the `oneOf([queryInputSchema, txInputSchema])`.
the `txBuildSchema` already checks for inputs array, if the array has length 1 with `type: "query"` it will be recognized as a query input, otherwise it will be a tx input array, which can be empty (meaning the wallet chooses the inputs).